### PR TITLE
NAS-105387 / 12.0 / Mount/Umount properly during iocage update failure

### PR DESCRIPTION
**Pull request by sonicaj**
Make sure to follow and check these boxes before submitting a PR! Thank you.

- [ ] Explain the feature
- [ ] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: iocage_lib/ioc_fetch.py [hunk 1]
```diff
@@ -944,6 +946,7 @@ def fetch_update(self, cli=False, uuid=None):
                         )
 
             finally:
+                su.Popen(['umount', f'{mount_root}/dev']).communicate()
```
Comment (william-gr) [inline iocage_lib/ioc_fetch.py:948]: Wont this try to umount twice if you hit the exceptions?

Comment (sonicaj) [reply iocage_lib/ioc_fetch.py:948]: No, `EXCEPTION` level means that we will exit this call for sure the first time it occurs, that's why the need to have these on multiple places

Comment (william-gr) [reply iocage_lib/ioc_fetch.py:948]: Are you sure? finally should run even if you raise an exception.

Comment (sonicaj) [reply iocage_lib/ioc_fetch.py:948]: Oh, okay the above one is unintended :) Good catch. I was thinking that we only have 2 umount commands - should've looked properly.

**Review by william-gr — APPROVED** [review]

**Pull request merged**